### PR TITLE
Add Hot-Reloading for Frontend

### PR DESCRIPTION
**Pull request by petabite**
### Administrative Info
Monday Board ID: 1181559666
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
- add `proxy` in frontend `package.json` to proxy API requests to the backend server. This allows us to utilize the React server's hot-reloading feature.
- edit README with new instructions (my vscode linted it too 😬 )
- update `npm start` script to start both frontend and backend servers using `concurrently`
- add `.DS_Store` to `.gitignore` for Mac users

### Testing
- tested that hot-reloading worked by changing a heading and watching the app reload with the new change
- tested that API requests were properly being proxied by: 
    - logging in with the test users
    - creating a sample recruitment application

### Confirmation of Change 

**Running `npm start` will now show the logs of both the frontend and backend servers**
image

### CODE REVIEW

**Review by hahsu — APPROVED** [review]
Is there a specific reason * were changed to -? If not, I think we should just leave them as *.  Otherwise it looks good to me. Thanks!

File: README.md [hunk 1]
```diff
@@ -20,26 +20,18 @@ The backend is written using a combination of NodeJS, Express, and MongoDB (the
...
-load some pages correctly, particularly with a dynamic URL component. It is recommended that you open
-a separate terminal to run the watcher:
-
-```
-npm run watch
-```
-
-However, if you are in a production environment, you can perform a one-time build of the frontend.
+If you are in a production environment, you can perform a one-time build of the frontend.
 This will not monitor the frontend directory for changes:
```
Comment (hahsu) [inline README.md:35]: Should this be changed since it does monitor the frontend directory for changes?

**Review by justinyaodu — APPROVED** [review]
LGTM!

**Comment by petabite** [comment]
> Is there a specific reason * were changed to -? If not, I think we should just leave them as *. Otherwise it looks good to me. Thanks!

Oh sorry, my vscode automatically lints Markdown whenever I save.

**Pull request merged**